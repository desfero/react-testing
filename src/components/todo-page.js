import './todo-page.css';
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';

import { changeTodoState, createTodo, removeTodo } from '../actions';
import {getTodos} from '../selectors';
import {TodoForm} from './todo-form';
import {TodoList} from './todo-list';


class TodoPageBase extends Component {
    static propTypes = {
        createTodo: PropTypes.func.isRequired,
        removeTodo: PropTypes.func.isRequired,
        changeTodoState: PropTypes.func.isRequired,
    };

    render() {
        return (
            <div id="todos">
                <TodoForm handleSubmit={this.props.createTodo}/>

                <div className="todo-list">
                    <TodoList
                        removeTodo={this.props.removeTodo}
                        changeTodoState={this.props.changeTodoState}
                        todos={this.props.todos}
                    />
                </div>
            </div>
        );
    }
}

const mapStateToProps = createSelector(
    getTodos,
    todos => ({todos})
);

const mapDispatchToProps = {
    removeTodo,
    createTodo,
    changeTodoState
};


export const TodoPage = connect(
    mapStateToProps,
    mapDispatchToProps
)(TodoPageBase);