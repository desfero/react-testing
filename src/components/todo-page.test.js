import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import { TodoPageBase } from './todo-page';

describe('todo-page', () => {
    it('should render component', () => {
        const nop = () => {};

        const wrapper = shallow(<TodoPageBase changeTodoState={nop} createTodo={nop} removeTodo={nop} todos={new List([])}/>);

        expect(wrapper).toMatchSnapshot();
    });
});