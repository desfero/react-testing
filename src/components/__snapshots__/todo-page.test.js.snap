// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPage should render the component 1`] = `
<div
  id="todos"
>
  <TodoForm
    handleSubmit={[Function]}
  />
  <button
    className="btn btn-outline-success"
    onClick={[Function]}
  >
    Done
  </button>
  <button
    className="btn btn-outline-danger"
    onClick={[Function]}
  >
    Todo
  </button>
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
  >
    All
  </button>
  <div
    className="todo-list"
  >
    <Component
      completeTodo={[Function]}
      removeTodo={[Function]}
      todos={
        Immutable.List [
        ]
      }
    />
  </div>
</div>
`;
